ls1intum/Artemis
# only lock again participations for future individual end dates in the unlock all operation

Invoking the unlock operation after the due date is over does not work, because all participations are immediately locked again.

### Checklist

### Description
The automatic lock is now only executed for future dates for exam exercises or exercises with manual review or exercises with build and test after due date.

### Steps for Testing
1. Create different kind of programming exercises
2. Try out the lock all and unlock all repositories actions in the programming exercise details page
3. Check that unlocking does not automatically lead to locking again. Locking should still appear in case the due date (or exam end date) is in the future, but should not happen automatically if it was in the past

<img width="907" alt="image" src="https://user-images.githubusercontent.com/744067/107120335-60e17d80-688d-11eb-8e09-3430f60b5651.png">


### Test Coverage
no change

### Screenshots
no changes
